lib/module.js

```diff
@@ -510,6 +510,10 @@ Module.prototype.load = function(filename) {
 Module.prototype.require = function(path) {
   assert(path, 'missing path');
   assert(typeof path === 'string', 'path must be a string');
+  // Ignore part of the path after null character if it exists
+  if (path.indexOf('\u0000') != -1) {
```

Comment:
I just ran a simple benchmark to compare the two on master and it seems `indexOf()` is still quite a bit faster overall.